rb/lib/selenium/webdriver/common/script.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+module Selenium
+  module WebDriver
+    class Script
+      def initialize(bridge)
+        @bidi = bridge.bidi
+        @log_entry_subscribed = false
+      end
+
+      # @return [int] ID of the handler
+      def add_console_message_handler
+        subscribe_log_entry unless @log_entry_subscribed
+        @bidi.add_callback('log.entryAdded') do |params|
+          yield(params) if params['type'] == 'console'
+        end
+      end
+
+      # @return [int] ID of the handler
```

Comment:
We want to follow this pattern for other languages too right?

Reply:
We can't do it in .NET because of the way the native event handlers work.

In Ruby this approach is especially useful because the norm is to work with blocks, which can't get assigned to a variable in a way that it can be used later to remove it.

Maybe it can come back with a type in Java to make it more clear what it is? What do you think?